BraveWallet/Crypto/BuySwapSwap/SwapCryptoView.swift

```diff
@@ -8,6 +8,35 @@ import BraveCore
 import BraveUI
 import struct Shared.Strings
 
+private struct RroundedBackgroundModifier: ViewModifier {
+  @Environment(\.sizeCategory) private var sizeCategory
+  
+  var backgroundColor: Color = Color(.secondaryBraveGroupedBackground)
+  
+  private var backgroundShape: some InsettableShape {
+    RoundedRectangle(cornerRadius: 10, style: .continuous)
+  }
+  
+  func body(content: Content) -> some View {
+    content
+      .background(
+        backgroundShape
+          .fill(backgroundColor)
+          .overlay(
+            // When using an accessibility font size, inset grouped tables automatically change to
+            // grouped tables with separators. So we will match this change and add a border around
+            // the buttons to make them appear more uniform with the table
+            Group {
+              if sizeCategory.isAccessibilityCategory {
+                backgroundShape
+                  .strokeBorder(Color(.separator))
+              }
+            }
+          )
+      )
+  }
+}
+
```

Comment:
Take out this `ViewModifier` since it's used in multiple views and components. Also added the ability to change the `.fill` color